### DISCUSSION TITLE
PDE loss function with multiple subnetworks

### DISCUSSION THREAD

**Discussion by leocotteleer** [lululxvi/deepxde#903] [Q&A]
Hi Prof @lululxvi, thanks for the very nice work!

I am trying to solve an inverse problem that estimates PDE coefficients from CFD data. First, I have created two simple neural nets to assimilate CFD data (values, derivatives etc).
Let's say:

```
net1 = dde.nn.FNN(layer_size1, activation1, initializer1)
model1 = dde.Model(data1, net1)

net2 = dde.nn.FNN(layer_size2, activation2, initializer2)
model2 = dde.Model(data2, net2)
```

Now I'd like to create a third network with a physics-informed loss function that takes into account the two first networks (net1 and net2):

```
def my_pde(x, y):
        y1 = model1.predict(x)
        y2 = model2.predict(x)

        d1 = ddde.grad.jacobian(y1, x, i=0, j=0)
        ...

data = dde.data.PDE(
    geometry,
    my_pde,
    [],
    num_domain=0,
    num_boundary=0,
)
```

However, it does not work as model.predict() is a Numpy call while x is a symbolic Tensor.

Do you have any ideas on how to proceed? I am working with the default backend _tensorflow.compat.v1_

Thanks in advance!

**Comment by lululxvi** [comment]
See examples at https://deepxde.readthedocs.io/en/latest/demos/pinn_forward.html and https://deepxde.readthedocs.io/en/latest/demos/pinn_inverse.html

**Reply by gnodking7** [reply]
Could you pinpoint the example that addresses this issue, please? I've looked through the examples but there does not seem to be a case where a trained model is used in a PDE for a different model. Thank you.

**Reply by gnodking7** [reply]
NVM, the inverse Lorenz problem with exogenous input example was exactly what I was looking for.
